src/main/java/org/teragrep/JavaPcre.java

```diff
@@ -132,10 +166,45 @@ public void pcre2_versioncheck(){
 //        compile_options = new LibJavaPcre.OptionsStruct();
 //    }
 
-    public void pcre2_compile_java(String pat, int size){
+    public void pcre2_gcontext_create(){
+        gcontext = LibJavaPcre.INSTANCE.pcre2_gcontext_create();
+    }
+    public void pcre2_ccontext_create(){
+        if (gcontext == null){
+            LOGGER.error("General context is not initialized properly!");
+        } else {
+            ccontext = LibJavaPcre.INSTANCE.pcre2_ccontext_create(gcontext);
+        }
+    }
+    public void pcre2_ccontext_set_extra_options(){
+        LibJavaPcre.INSTANCE.pcre2_ccontext_set_extra_options(ccontext, extra_options);
+    }
+    public void pcre2_mcontext_create(){
+        if (gcontext == null){
+            LOGGER.error("General context is not initialized properly!");
+        } else {
+            mcontext = LibJavaPcre.INSTANCE.pcre2_mcontext_create(gcontext);
+        }
+    }
+    public void pcre2_gcontext_free(){
+        LibJavaPcre.INSTANCE.pcre2_gcontext_free(gcontext);
+        gcontext = null;
+    }
+    public void pcre2_ccontext_free(){
+        LibJavaPcre.INSTANCE.pcre2_ccontext_free(ccontext);
+        ccontext = null;
+    }
+    public void pcre2_mcontext_free(){
+        LibJavaPcre.INSTANCE.pcre2_mcontext_free(mcontext);
+        mcontext = null;
+    }
+
+    public void pcre2_compile_java(String pat){
         pattern = pat;
-        pattern_size = size;
-        re = LibJavaPcre.INSTANCE.pcre2_jcompile(pattern, pattern_size, compile_options);
+        re = LibJavaPcre.INSTANCE.pcre2_jcompile(pattern, pattern_size, compile_options, ccontext);
+        if (re == null){
+            LOGGER.error("Compiling of the match pattern failed.");
```

Comment:
Java uses https://docs.oracle.com/javase/7/docs/api/java/util/regex/PatternSyntaxException.html for these, so I think it would be preferrable to have matching one

Reply:
Added the exception to the code